File: app/src/main/java/com/demo/nytimesnews/di/MainAppComponent.java
package com.demo.nytimesnews.di;

import android.app.Application;

import com.demo.nytimesnews.di.module.ContextModule;
import com.demo.nytimesnews.di.module.NetworkModule;
import com.demo.nytimesnews.di.module.NewsListModule;
import com.demo.nytimesnews.di.qualifier.RetrofitQualifier;
import com.demo.nytimesnews.di.scope.ApplicationScope;
import com.demo.nytimesnews.feature.news.list.NewsListActivity;

import dagger.Component;
import retrofit2.Retrofit;

/**
 * Created by Rashida on 4/9/19.
 */

@ApplicationScope
@Component(modules = {
        ContextModule.class,
        NetworkModule.class,
        NewsListModule.class
})
public interface MainAppComponent {

    Application getApplication();

    @RetrofitQualifier
    Retrofit retrofitNews();

    void inject(NewsListActivity newsListActivity);

}

File: app/src/main/java/com/demo/nytimesnews/di/MainAppComponent.kt
package com.demo.nytimesnews.di;

import com.demo.nytimesnews.di.module.ContextModule;
import com.demo.nytimesnews.di.module.NetworkModule;
import com.demo.nytimesnews.di.module.NewsListModule;
import com.demo.nytimesnews.di.scope.FeatureScope;
import com.demo.nytimesnews.feature.news.list.NewsListActivity;

import dagger.Component;

/**
 * Created by Rashida on 4/9/19.
 */

@FeatureScope
@Component(modules = {
        ContextModule.class,
        NetworkModule.class,
        NewsListModule.class
})
public interface MainAppComponent {

    void inject(NewsListActivity newsListActivity);

}
